ThunderCatsTests/UIImageTests.m

```diff
@@ -0,0 +1,497 @@
+//
+//  UIImageTests.m
+//  ThunderCats
+//
+//  Created by William Grand on 5/11/16.
+//  Copyright © 2016 Metova. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "UIImage+TCAdditions.h"
+#import "UIImageExtensionHelper.h"
+
+@interface UIImageTests : XCTestCase
+
+@end
+
+@interface MockAssetRepresentation: ALAssetRepresentation
+
+@end
+
+@implementation MockAssetRepresentation
+
+- (CGImageRef)fullResolutionImage {
+    return [UIImage tc_imageWithColor:[UIColor redColor]].CGImage;
+}
+
+@end
+
+@interface MockAsset: ALAsset
+
+@end
+
+@implementation MockAsset
+
+- (ALAssetRepresentation *)defaultRepresentation {
+    return [[MockAssetRepresentation alloc] init];
+}
+
+
+@end
+
+@interface MockImage: UIImage
+
+@end
+
+@implementation MockImage
+
+UIImageOrientation mockedOrientation;
+
+- (UIImage *)initWithCGImage:(CGImageRef)cgImage orientation:(UIImageOrientation)orientation {
+    mockedOrientation = orientation;
+    return [super initWithCGImage:cgImage];
+}
+
+- (UIImageOrientation)imageOrientation {
+    return mockedOrientation;
+}
+
+@end
+
+@implementation UIImageTests
+
+- (void)setUp {
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+- (void)testImageWithColor {
+    
+    CGFloat inRed = 136.0/255.0;
+    CGFloat inGreen = 250.0/255.0;
+    CGFloat inBlue = 70.0/255.0;
+    CGFloat inAlpha = 1.0;
+    
+    UIImage *image = [UIImage tc_imageWithColor:[UIColor colorWithRed:inRed green:inGreen blue:inBlue alpha:inAlpha]];
+    
+    XCTAssertNotNil(image);
+    XCTAssertTrue([image isKindOfClass:[UIImage class]]);
+    
+    CGPoint pixelPoint = CGPointMake(0, 0);
+    
+    CGDataProviderRef provider = CGImageGetDataProvider(image.CGImage);
+    CFDataRef pixelData = CGDataProviderCopyData(provider);
+    const UInt8* data = CFDataGetBytePtr(pixelData);
+    
+    int numberOfColorComponents = 4; // R,G,B, and A
+    float x = pixelPoint.x;
+    float y = pixelPoint.y;
+    float w = image.size.width;
+    int pixelInfo = ((w * y) + x) * numberOfColorComponents;
+    
+    UInt8 blue = data[pixelInfo];
+    UInt8 green = data[(pixelInfo + 1)];
+    UInt8 red = data[pixelInfo + 2];
+    UInt8 alpha = data[pixelInfo + 3];
+    CFRelease(pixelData);
+    
+    // RGBA values range from 0 to 255
+    UIColor *pixelColor = [UIColor colorWithRed:red/255.0
+                           green:green/255.0
+                            blue:blue/255.0
+                           alpha:alpha/255.0];
+    
+    CGFloat outRed, outGreen, outBlue, outAlpha;
+    [pixelColor getRed:&outRed green:&outGreen blue:&outBlue alpha:&outAlpha];
+    
+    XCTAssertEqual(inRed, outRed);
```

Comment:
Because these are floating point numbers, we need to use `XCTAssertEqualWithAccuracy`.